processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceMethod.java

```diff
@@ -263,7 +265,7 @@ private SourceMethod(Builder builder, MappingMethodOptions mappingMethodOptions)
     private boolean determineIfIsObjectFactory() {
         boolean hasNoSourceParameters = getSourceParameters().isEmpty();
         boolean hasNoMappingTargetParam = getMappingTargetParameter() == null;
-        return !isLifecycleCallbackMethod() && !returnType.isVoid()
+        return !isLifecycleCallbackMethod() && !returnType.isVoid() && !returnType.isPrimitive()
```

Comment:
I think I can understand why you added `!returnType.isPrimitive()`.. But is it strictly part of this PR? Should there be a separate testcase validating this?

Reply:
This is the one I wasn't happy about. It is too hackish to me since its not explicit. I was wondering if it would be better to invoke `determineIfIsPresenceCheck` instead of `!returnType.isPrimitive()`, but then the method leveling is kinda wrong.
I would need some consulting from your behalf on this...

Reply:
I came across execution when object factory method was detected instead of presence check. What I assumed was: object factory can't return a primitive ... but that is implicit ... and that's why I'm not happy with that IF

Reply:
For this to be the same as in the rest of the flow wouldn't it make more sense to say here:
```java
boolean hasNoTargetPropertyNameParam = getTargetPropertyNameParameter() == null;
return !isLifecycleCallbackMethod() && !returnType.isVoid()
            && hasNoMappingTargetParam && hasNoTargetPropertyNameParam
            && ( hasObjectFactoryAnnotation || hasNoSourceParameters );
```
If I correctly understand that you want to exclude methods that have a `@TargetPropertyName` from being seen as an object factory?

Reply:
Yep, That's what I was looking for! Thanks!